js/jquery.dbRow.js

```diff
@@ -168,15 +172,27 @@
 		this.destroy();
 	    }
 	},
-	actionReturnError: function(action,errorMessage, errorType) {
-            if ( errorType == 'NAVIGATION' ) {
+	actionReturnError: function(action,errorMessage, errorType, jqXHR) {
+            // Handler for errors returned from server.
+            switch(errorType) {
+            case "NAVIGATION":
                 return;
+            case "HTTP":
+                var xml = $.parseXML(jqXHR.responseText);
+                var xmlError = $(xml).find('error').first();
+                if ( xmlError.length == 1 ) {
+                    this.error = xmlError.text();
+                }
+                break;
+            default:
+                this.error = errorMessage;
             }
-	    this.error = errorMessage;
-	    this.setState('error');
-	    if ( errorType != 'USER' ) {
-		qcode.alert(errorMessage);
-	    }
+
+            // Alert on all errors that aren't user errors.
+            if ( jqXHR.status !== 400 ) {
+                qcode.alert(this.error)
```

Comment:
The previous code didn't want to alert when there was a `USER` error (because of doubling up on errors I think? This was the case in my testing on MLA ERP but not sure if it would be the same on other projects). Since we've moved these errors over to using HTTP status codes then the only way to determine a `USER` error is through the status code however the status code 400 is also used by `BAD_REQUEST`.

I think it's okay to not alert for either `USER` or `BAD_REQUEST` errors although it is something to be aware of to potentially look into in the future. A possible fix would be to use a different status code for `USER` errors but deciding upon which codes are appropriate is the challenge.